## This script analyzes the peaks generated by the CHIP-seq analysis and
## defines the genes that are regulated by the transcription factor of study.
## Besides, this script performs GO enrichment of the regulated genes using KEGG
## tools and creates metabolics pathways using KEGG pathview.

args<- commandArgs(trailingOnly=T)


promoter.length <- as.numeric(args[[1]])
output.file.name<- args[[2]]
peakfile<-args[[3]]
library(ChIPseeker)
library(TxDb.Athaliana.BioMart.plantsmart28)
txdb <- TxDb.Athaliana.BioMart.plantsmart28


## Read peaks file
peaks <- readPeakFile(peakfile = peakfile,header=FALSE)

## Define promoter region
promoter <- getPromoters(TxDb=txdb, 
                         upstream=promoter.length, 
                         downstream=promoter.length)

## Peaks annotation
peakAnno <- annotatePeak(peak = peaks, 
                             tssRegion=c(-promoter.length, promoter.length),
                             TxDb=txdb)
## Pie plot of peaks annotation
jpeg(filename="plotAnnoPie.jpeg")
plotAnnoPie(peakAnno)
dev.off()

## Distribution of genomic loci relative to TSS
jpeg(filename="plotDistToTSS.jpeg")
plotDistToTSS(peakAnno,
              title="Distribution of genomic loci relative to TSS",
              ylab = "Genomic Loci (%) (5' -> 3')")
dev.off()

## Turn peaks annotation into data frame
peak.annotation <- as.data.frame(peakAnno)
head(peak.annotation)
## Selection of genes regulated by the promoter of study
target.genes <- peak.annotation$geneId[peak.annotation$annotation == "Promoter"]
write(x = target.genes,file = output.file.name)

## KEGG analysis

library(clusterProfiler)
library(AnnotationDbi)
library(org.At.tair.db)
## GO enrichment
## Get all arabidopsis thaliana genes names
ath.genes<-as.data.frame(genes(txdb))
ath.genes.id<-ath.genes$gene_id
## GO enrichment for biological process
enrichGOgenes.bp<-enrichGO(target.genes,universe= ath.genes.id,OrgDb = "org.At.tair.db",keyType = "TAIR",ont = "BP")
## GO enrichment for molecular function
enrichGOgenes.mf<-enrichGO(target.genes,universe= ath.genes.id,OrgDb = "org.At.tair.db",keyType = "TAIR",ont = "MF")
## GO enrichment for cellular compartiment
enrichGOgenes.cc<-enrichGO(target.genes,universe= ath.genes.id,OrgDb = "org.At.tair.db",keyType = "TAIR",ont = "CC")

library(topGO)
library(Rgraphviz)
## Barplots of the previous GO enrichment analysis
jpeg(filename = "barplotGO_bp.jpeg")
barplot(enrichGOgenes.bp)
dev.off()
jpeg(filename = "barplotGO_mf.jpeg")
barplot(enrichGOgenes.mf)
dev.off()
jpeg(filename = "barplotGO_cc.jpeg")
barplot(enrichGOgenes.cc)
dev.off()

## KEGG photosyntesis pathway
library(pathview)
## Create a universe indicating the target genes
my.universe<-rep(0,length(ath.genes.id))
names(my.universe)<-ath.genes.id
my.universe[target.genes]<-1
## Creating the KEGG pathway
keggpathway<-pathview(gene.data=my.universe, species = "ath", pathway.id = "ath00195", gene.idtype = "TAIR")

